### PR TITLE
Export files inside nested directories

### PR DESCRIPTION
**Pull request by coder6583**
## Description
- Export files inside nested directories when exporting assessments and courses
- Filters out the handin directory when exporting

## Motivation and Context
- When the course had directories that had nested files, the files weren't showing up. 
image

↑ A nested file in the course directory
image

↑ No file

## How Has This Been Tested?
- Created an assessment with a writeup directory containing a pdf file
- Handed in a few files to the assessment to create files inside the handin directory
- Exported assessment and course with assessments

## Types of changes
- [x] Bug fix (non-breaking change which fixes an issue)

## Checklist:
- [x] I have run rubocop and erblint for style check. If you haven't, run `overcommit --install && overcommit --sign` to use pre-commit hook for linting

## Other issues / help required
Should I use Find.find over Dir[].each?

If unsure, feel free to submit first and we'll help you along.

### CODE REVIEW

**Comment by KesterTan** [comment]
I think in this case, it should be fine to not use Find.find since you're already disabling handins

**Comment by KesterTan** [comment]
Looks good so far, but I think we can perhaps consider adding a toggle so that people can choose if they want to download handins or not.

File: app/controllers/assessments_controller.rb [hunk 1]
```diff
@@ -467,19 +467,8 @@ def export
...
-          if File.directory?(file)
-            tar.mkdir relative_path, mode
-          elsif !relative_path.starts_with? File.join(@assessment.name,
-                                                      @assessment.handin_directory)
-            tar.add_file relative_path, mode do |tarFile|
-              File.open(file, "rb") { |f| tarFile.write f.read }
-            end
-          end
-        end
+        filter = [File.join(dir_path, asmt_dir, @assessment.handin_directory)]
```
Comment (damianhxy) [inline app/controllers/assessments_controller.rb:470]: Nit: You can use `@assessment.handin_directory_path` instead

File: app/models/course.rb [hunk 2]
```diff
@@ -340,19 +340,8 @@ def generate_tar(export_configs)
...
-              if File.directory?(file)
-                tar.mkdir relative_path, mode
-              elsif !relative_path.starts_with? File.join(asmt_dir,
-                                                          assessment.handin_directory)
-                tar.add_file relative_path, mode do |tar_file|
-                  File.open(file, "rb") { |f| tar_file.write f.read }
-                end
-              end
-            end
+            filter = [File.join(base_path, asmt_dir, assessment.handin_directory)]
```
Comment (damianhxy) [inline app/models/course.rb:343]: Similarly here, use `handin_directory_path`

**Review by KesterTan — APPROVED** [review]
LGTM

**Pull request merged**